### DISCUSSION TITLE
Formatting LocalDate and LocalDateTime

### DISCUSSION THREAD

**Discussion by rclarke1** [Blazebit/blaze-persistence#1494] [Q&A] [answered]
Hi,
This looks like a great project.
Previously I was using @JsonFormat to format dates when returning data through a controller. 
How can I format a LocalDate to yyyy-mm-dd when returning an EntityView through a controller.
Thanks

**Comment by beikov** [comment, marked as answer]
Hey there. You can still use `@JsonFormat` if you like. Otherwise, you can also use the `DateTimeFormatter` API like this in your entity views:

```java
@EntityView(MyEntity.class)
public interface MyEntityView {

    @Mapping("date")
    @JsonIgnore
    LocalDate getDateAsObject();

    default String getDate() {
        return DateTimeFormatter.ISO_LOCAL_DATE.format(getDateAsObject());
    }
}
```

**Reply by rclarke1** [reply]
Great thanks, putting @JsonFormat on the EntityView Interface worked.
